### PR TITLE
DDF-5395 Cleanup after itests instead of resetting the system after every class

### PR DESCRIPTION
**Pull request by brjeter**
#### What does this PR do?
Currently, the whole container resets after each itest class. Instead, the tests now properly clean up afterwards to save some time. I've seen about 3-4 minutes shaved off due to this change.

#### Who is reviewing it? 
@clockard
@coyotesqrl 
@pklinef 
@shaundmorris
@stustison 
@AzGoalie 

#### Select relevant component teams: 
@codice/build 
@codice/continuous-integration 

#### How should this be tested?
CI

#### What are the relevant tickets?
Fixes: #5395 

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
Build now

**Comment by brjeter** [comment]
build now

File: libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java [hunk 1]
```diff
@@ -128,8 +141,6 @@ private void failRule(String testClassName, Throwable throwable, String failureM
   }
 
   private void finished(Description description) {
-    LOGGER.info("Finished {} ({}/{})", description.getMethodName(), testsExecuted, testCount);
```
Comment (pklinef) [inline libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java]: ❓ Why was this removed instead of being moved to debug or trace?

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 2]
```diff
@@ -1003,6 +1040,7 @@ public void testConnectedSourceStatus() throws Exception {
 
     } finally {
       getSecurityPolicy().configureRestForGuest();
+      cleanupConnectedSources();
```
Comment (shaundmorris) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: ❔ do we still have connected sources?

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 3]
```diff
@@ -2185,11 +2232,17 @@ private void verifyEvents(
   }
 
   private void setupConnectedSources() throws IOException {
-    getServiceManager()
-        .createManagedService(
-            CSW_CONNECTED_SOURCE_FACTORY_PID,
-            getCswConnectedSourceProperties(
-                CONNECTED_SOURCE_ID, CSW_PATH.getUrl(), getServiceManager()));
+    connectedPid =
```
Comment (shaundmorris) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: This answers my question on Connected Sources. Might be out scope for this PR but would like to remove all connected source itests, maybe completely.

File: libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java [hunk 4]
```diff
@@ -105,6 +107,17 @@ private void starting(Description description) {
       failRule(testClassName, throwable, BEFORE_EXAM_FAILURE_MESSAGE);
     }
 
+    if (firstSuiteRun) {
+      try {
+        firstSuiteRun = false;
+        runAnnotations(BeforeSuite.class, testClass);
+      } catch (Throwable throwable) {
+        setupFailed = true;
+        LOGGER.error("Failed to setup SUITE for " + testClassName, throwable);
```
Comment (shaundmorris) [inline libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java]: ❔ should we be using parameterized logging here? Can't remember on error messages.

**Comment by rzwiefel** [comment]
Should we writeup a little "What to make sure to do when you write/update iTests" blurb that we can throw somewhere nearby (perhaps in itest readme or such)? That way we can hopefully make sure anyone in the future does these same things also.

File: libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java (hunk 1 above)
Comment (brjeter) [reply libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java]: Because of the mostly redundant logs that are now produced by the `ExamResultLogger`:
https://github.com/codice/ddf/pull/5396/files#diff-d79256616b2290cfe3b63bfe7877fcabR25-R32

File: distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java [hunk 5]
```diff
@@ -597,6 +611,8 @@ private static Integer findPortNumber(Integer portToTry) {
     return options(
         editConfigurationFilePut(
             LOGGER_CONFIGURATION_FILE_PATH, "log4j2.rootLogger.level", globalLogLevel),
+        // Always print the start/stop/result of individual test methods
+        composite(createSetLogLevelOption("org.codice.ddf.test.common.annotations", "INFO")),
```
[the remark this answers is not in the document]
Comment (brjeter) [reply distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/AbstractIntegrationTest.java]: Yeah the default for the global log level is WARN for that reason. The amount of logs that this option adds are just a tad more than 2 per test method, so it's not too bad.

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 6]
```diff
@@ -282,6 +307,18 @@ public void beforeExam() throws Exception {
     }
   }
 
+  @AfterExam
+  public void afterExam() throws Exception {
+    getServiceManager().stopManagedService(openSearchPid);
```
[the remark this answers is not in the document]
Comment (brjeter) [reply distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: It's possible, so I will do a `null` check in all of these scenarios.

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 7]
```diff
@@ -1560,50 +1598,54 @@ public void testEnterpriseSearch() throws Exception {
...
-            Condition.parameter("id", metacardId));
+      cswServer
+          .verifyHttp()
+          .times(
+              1,
+              Condition.uri("/services/csw"),
+              Condition.parameter("request", "GetRecordById"),
+              Condition.parameter("id", metacardId));
+    } finally {
+      getCatalogBundle().setupCaching(false);
```
[the remark this answers is not in the document]
Comment (brjeter) [reply distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: I didn't think so because only 3 out of 20+ tests in this class actually mess with this configuration.

File: libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java (hunk 4 above)
Comment (brjeter) [reply libs/test-common/src/main/java/org/codice/ddf/test/common/annotations/PaxExamRule.java]: We can't in this case because we're passing in the `Throwable`.

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
squashed and rebased.
build now

**Comment by brjeter** [comment]
`The download was interrupted; unable to complete the update`
build now
